### PR TITLE
Update MainActivity.kt

### PR DESCRIPTION
**Pull request by kenkelbt**
Moved variables into MainActivity for encapsulation, set firebaseUser to the current user, and split up the onCreate function as it had a grouping of functions setting up the screen which would make more sense as it's own variable. I had these updates separated into three commits but I wasn't able to create a branch directly to this GitHub because it said I didn't have write permissions. For that reason, I just copied over my changes to a fork I created of the repository and it joined all my commits into one. In terms of the analysis of the program, there really isn't a whole lot to change because it's very well written. The program was available on GitHub in time, I'm just having issues accessing it now, so I'll have to reach out to the team to see what's going on. The program was documented decently, I would recommend a little more documentation. The program does compile. I made my changes to reduce technical debt by utilizing encapsulation and splitting up functions.

Three Commits on My Repo:
https://github.com/grisbyar/RoomRemix2/pull/5/commits/ceaaf1dc4872fcaf10f3aa154c140712bbc527ee
https://github.com/grisbyar/RoomRemix2/pull/5/commits/5b5bfdad3c24287ad67c6ca6c45736ab9118735a
https://github.com/grisbyar/RoomRemix2/pull/5/commits/5d8f6fc2d893ae292a88a20c4e450953753aab06

Here are some examples of what I've learned in this code review.

Coroutines:
The use of coroutines to handle asynchronous tasks in a more efficient way than using callbacks or threads. For example, in this code snippet, coroutines are used to execute network requests and update the UI accordingly:

viewModelScope.launch {
    try {
        val response = repository.getData()
        data.value = response
    } catch (e: Exception) {
        // Handle exception
    }
}

ViewModel:
The use of ViewModel to manage the UI-related data in a lifecycle-aware way. For example, in this code snippet, the ViewModel is used to store the data that is displayed in the UI:

class MyViewModel(private val repository: MyRepository) : ViewModel() {
    val data = MutableLiveData<List<MyData>>()
    
    init {
        loadData()
    }
    
    private fun loadData() {
        viewModelScope.launch {
            try {
                val response = repository.getData()
                data.value = response
            } catch (e: Exception) {
                // Handle exception
            }
        }
    }
}

Lambda expressions:
The use of lambda expressions to simplify code and make it more concise. For example, in this code snippet, a lambda expression is used to filter a list of data based on a certain condition:

val filteredData = data.filter { it.value > 0 }

### CODE REVIEW

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 1]
```diff
@@ -35,20 +35,21 @@ import com.ideastorm.v25001.dto.User
...
     var selectedParticipantOption: String? = null
     var selectedTypeOption: String? = null
     var selectedPriceOption: String? = null
-    private val viewModel: MainViewModel by viewModel<MainViewModel>()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             viewModel.fetchActivity()
+            val firebaseUser = FirebaseAuth.getInstance().currentUser
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:52]: I don't recommend this change, as this is a shadow condition: two variables with the same name, different scope.

The other variable is declared on line 42:
 var firebaseUser: FirebaseUser? = FirebaseAuth.getInstance().currentUser

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 2]
```diff
@@ -35,20 +35,21 @@ import com.ideastorm.v25001.dto.User
 import com.ideastorm.v25001.ui.theme.IdeaStormTheme
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
-
 /**
  * This class represents the main activity for the IdeaStorm app and sets up the UI layout and theme
  */
 class MainActivity : ComponentActivity() {
     var firebaseUser: FirebaseUser? = FirebaseAuth.getInstance().currentUser
+    private val viewModel: MainViewModel by viewModel<MainViewModel>()
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:43]: It looks like this was moved up 4 lines... I'm not clear on the tech debt reduction from this change.

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 3]
```diff
@@ -60,17 +61,22 @@ class MainActivity : ComponentActivity() {
                     modifier = Modifier.fillMaxWidth(),
                     color = MaterialTheme.colors.background
                 ) {
+                    IdeaStormScreen()
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:64]: Careful here, it looks like this was moved inside a } when compared to what it is replacing.  Is this in the correct location?

**Comment by discospiff** [comment]
There is not significant tech debt reduction in this branch, so I don't recommend merging it.

**Comment by steelesh** [comment]
Closing pull request as it doesn't reduce technical debt.

**Pull request closed without merging**